redis_collections/dicts.py

```diff
@@ -516,8 +518,6 @@ def op_trans(pipe):
                 other_counter = collections.Counter(
                     {k: v for k, v in other.iteritems(pipe=pipe)}
                 )
-            elif isinstance(other, RedisCollection):
```

Comment:
The Python `Counter` [doesn't really care](https://github.com/python/cpython/blob/3.5/Lib/collections/__init__.py#L693-L752) about non-`Counter` instances, so this branch never gets a change to execute.